contracts/bonding/BondingManager.sol

```diff
@@ -884,21 +835,78 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         return delegators[_delegator].unbondingLocks[_unbondingLockId].withdrawRound > 0;
     }
 
+    function increaseTranscoderTotalStake(address _transcoder, uint256 _amount) internal {
```

Comment:
Currently, whenever we increase a delegator's `delegatedAmount` field, we also check if the delegate is a registered transcoder and if so we call `increaseTranscoderTotalStake()`. But, given how `increaseTranscoderTotalStake()` is implemented we need to make sure that `delegatedAmount` is increased *after* `increaseTranscoderTotalStake()` or else we will count `_amount` twice when updating active stake.

I see a few things that are not problems right now, but we should be careful about:

- Making sure to always increase `delegatedAmount` *after* calling `increaseTranscoderTotalStake()`
- Making sure to always update `delegatedAmount` when we call `increaseTranscoderTotalStake()` and vice-versa

One way that we can reduce the chance of error is to:

- Change the name of `increaseTranscoderTotalStake()` to `increaseTotalStake()` (I'm going simple with the naming)
- Move the logic for increasing the `delegatedAmount` field on a `Delegator` struct into `increaseTotalStake()`
- Move the logic for checking if an address is a registered transcoder into this function and execute the current function body only if the address is a registered transcoder

WDYT?

Reply:
I agree that the current flow isn't optimal and could lead to dev errors. 

I contemplated adding increasing delegatedAmount into those methods but didn't add it since all delegator state updates were happening outside of the scope of those increase/decrease helpers.

I'm not against adding it there however, on the contrary. So the feedback is much appreciated.